python/ray/serve/_private/logging_utils.py

```diff
@@ -161,6 +166,33 @@ def get_component_logger_file_path() -> Optional[str]:
                 return absolute_path[len(ray_logs_dir) :]
 
 
+original_makeRecord = logging.Logger.makeRecord
+
+
+def make_record_with_extra(
+    self, name, level, fn, lno, msg, args, exc_info, func=None, extra=None, sinfo=None
+):
+    # Unfortunately logging doesn't provide any functionality to differentiate extra
+    # fields from "record" object. So we add _extra field by ourselves to record the
+    # extra field.
+
+    record = original_makeRecord(
+        self,
+        name,
+        level,
+        fn,
+        lno,
+        msg,
+        args,
+        exc_info,
+        func=func,
+        extra=extra,
+        sinfo=sinfo,
+    )
+    record._extra = extra
```

Comment:
Should we `deepcopy` `extra` here in case either Serve or the `logging` package mutates the `extra` dict in the future?

Reply:
not applied, updated pr without using "_extra"